runtime/tests/port/j9strTest.c

```diff
@@ -1534,40 +1534,46 @@ j9str_WinacpToMutf8(struct J9PortLibrary *portLibrary)
  * @param[in] buffer - pointer to the buffer intended for atoe_vsnprintf()
  * @param[in] bufferLength - the buffer length
  * @param[in] expectedOutput - The expected string
+ * @param[in] expectedCount - The expected response from atoe_vsnprintf()
  * @param[in] format - The string format
  *
  * @return TEST_PASS on success
  *         TEST_FAIL on error
  */
 static int
-printToBuffer(struct J9PortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, const char *format, ...)
+printToBuffer(struct J9PortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, int expectedCount, const char *format, ...)
 {
 	PORT_ACCESS_FROM_PORT(portLibrary);
 	int stringLength = 0;
+	int expectedLength = 0;
+	int actualCount = 0;
 	va_list args;
 
 	va_start(args, format);
-	stringLength = atoe_vsnprintf(buffer, bufferLength, format, args);
-	va_end (args);
+	actualCount = atoe_vsnprintf(buffer, bufferLength, format, args);
+	expectedLength  = (NULL == expectedOutput) ? 0 : strlen(expectedOutput);
+	stringLength = (NULL == buffer) ? 0 : strlen(buffer);
+	va_end(args);
 
-	if (stringLength >= 0) {
+	if (actualCount >= 0) {
 		if (precision) {
 			j9tty_printf(PORTLIB, "\n\tFinish Testing: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
 		} else {
 			j9tty_printf(PORTLIB, "\n\tFinish Testing without precision specifier: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
 		}
 
-		if ((stringLength == strlen(expectedOutput))
+		if ((stringLength == expectedLength)
+			&& (expectedCount == actualCount)
 			&& (0 == memcmp(buffer, expectedOutput, stringLength))
```

Comment:
This call to `memcmp()` must not occur if `buffer` is `NULL`.

Reply:
I believe the `stringLength` variable should take care of that. it would be zero if the strings are null and memcmp would return 0

Reply:
It might, but I don't think it's reasonable to make such an assumption.

Reply:
used non null string since I created it for the purpose of printing